src/blob/index/bptree/core.rs

```diff
@@ -385,37 +394,37 @@ where
         Ok(())
     }
 
-    fn hash_valid(header: &IndexHeader, buf: &mut Vec<u8>) -> Result<bool> {
+    fn hash_valid(header: &IndexHeader, buf: &mut [u8]) -> Result<bool> {
         let hash = header.hash.clone();
         let mut header = header.clone();
         header.hash = vec![0; ring::digest::SHA256.output_len];
         header.set_written(false);
-        serialize_into(buf.as_mut_slice(), &header)?;
+        serialize_into(&mut buf[..], &header)?;
```

Comment:
`buf` is already a reference to mut slice:
```suggestion
        serialize_into(buf, &header)?;
```

Reply:
It will append if not used with this syntax